homeassistant/components/kraken/__init__.py

```diff
@@ -0,0 +1,163 @@
+    DEFAULT_SCAN_INTERVAL,
+    DEFAULT_TRACKED_ASSET_PAIR,
+    DISPATCH_CONFIG_UPDATED,
+    DOMAIN,
+)
+from .utils import get_tradable_asset_pairs
+
+PLATFORMS = ["sensor"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the kraken component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up kraken from a config entry."""
+    kraken_data = KrakenData(hass, config_entry)
+    await kraken_data.async_setup()
+    hass.data[DOMAIN] = kraken_data
+    config_entry.add_update_listener(async_options_updated)
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(config_entry, "sensor")
+    )
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(config_entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+    if unload_ok:
+        for unsub_listener in hass.data[DOMAIN].unsub_listeners:
+            unsub_listener()
+        hass.data.pop(DOMAIN)
+
+    return unload_ok
+
+
+class KrakenData:
+    """Define an object to hold kraken data."""
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
+        """Initialize."""
+        self._hass = hass
+        self._config_entry = config_entry
+        self._api = pykrakenapi.KrakenAPI(krakenex.API(), retry=0, crl_sleep=0)
+        self.tradable_asset_pairs = None
+        self.coordinator = None
+        self.unsub_listeners = []
+
+    async def async_update(self) -> None:
+        """Get the latest data from the Kraken.com REST API.
+
+        All tradeable asset pairs are retrieved, not the tracked asset pairs
+        selected by the user. This enables us to check for an unknown and
+        thus likely removed asset pair in sensor.py and only log a warning
+        once.
+        """
+        try:
+            async with async_timeout.timeout(10):
+                return await self._hass.async_add_executor_job(self._get_kraken_data)
+        except pykrakenapi.pykrakenapi.KrakenAPIError as error:
+            if "Unknown asset pair" in str(error):
+                _LOGGER.info(
+                    "Kraken.com reported an unknown asset pair. Refreshing list of tradable asset pairs."
+                )
+                await self._async_refresh_tradable_asset_pairs()
+            else:
+                raise UpdateFailed(
+                    f"Unable to fetch data from Kraken.com: {error}"
+                ) from error
+        except pykrakenapi.pykrakenapi.CallRateLimitError:
+            _LOGGER.warning(
+                "Exceeded the Kraken.com call rate limit. Increase the update interval to prevent this error."
```

Comment:
Do we hit it faster because we always request all possible tradable asset pairs?

Reply:
No it is measured by api calls not by requested asset pairs.